e2e/nomostest/nt.go

```diff
@@ -1006,6 +1017,26 @@ func gitCommitFromSpec(nt *NT, gitSpec *v1beta1.Git) (string, error) {
 	return columns[0], nil
 }
 
+// cloneCloudSourceRepo clones the provided Cloud Source Repository to a local
+// temp directory and returns the directory path.
+// This special logic is needed to handle authentication to the repo in CI.
+func cloneCloudSourceRepo(nt *NT, repo string) (string, error) {
+	repoName := path.Base(repo)
+	cloneDir := path.Join(nt.TmpDir, "csr-repos", repoName)
+	// return if directory was already created
```

Comment:
You're ahead of me. I was going to left a comment for the existence check and it just shows up after refresh. 😄

Reply:
Suggest namespacing the directory by project too, not just repo name. Otherwise if you run locally there’s a chance it’s configured for a different project.

Reply:
`nt.TmpDir` is scoped to a single test case, and each test case is scoped to a single project (set by the env var). So I don't think this should be an issue, but it wouldn't hurt

Reply:
👍